### PR TITLE
feat: improved Creepy Girl and Polar Bear logic, President rework

### PR DESCRIPTION
**Pull request by Golbolco**
cg can't win if dead, polar bears win on achieving majority

president no longer kills all villagers on death. works like on EM (1 prez death = mafia win). didn't edit the linchpin card though

### CODE REVIEW

File: Games/types/Mafia/roles/Village/President.js [hunk 1]
```diff
@@ -4,6 +4,26 @@ module.exports = class President extends Role {
   constructor(player, data) {
     super("President", player, data);
     this.alignment = "Village";
-    this.cards = ["VillageCore", "WinWithVillage", "KillAlignedOnDeath"];
+    this.cards = ["VillageCore", "WinWithVillage",];
+    this.listeners = {
+      roleAssigned: function (player) {
```
Comment (pikulet) [inline Games/types/Mafia/roles/Village/President.js:9]: is the purpose of this just to announce the president's modifiers? r9 made a function that can be used here

File: Games/types/Mafia/roles/cards/Polariser.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ module.exports = class Polariser extends Card {
     this.meetings = {
       Polarise: {
         states: ["Night"],
-        flags: ["voting", "multi", "group", "speech"],
+        flags: ["group", "speech", "voting", "multiActor"],
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/Polariser.js:11]: did you want it to be both multi and multiactor?

File: Games/types/Mafia/roles/cards/WinFourPolarisedDeaths.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ module.exports = class WinFourPolarisedDeaths extends Card {
         if (
           this.player.alive &&
           !winners.groups[this.name] &&
-          (this.polarisedKills >= 4 || aliveCount == 2)
+          (this.polarisedKills >= 4 || counts["Polar Bear"] >= aliveCount / 2 && aliveCount > 0)
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/WinFourPolarisedDeaths.js:15]: counts is only for alignments not roles

File: Games/types/Mafia/roles/cards/WinWithMafia.js [hunk 4]
```diff
@@ -50,6 +51,25 @@ module.exports = class WinWithMafia extends Card {
           );
         }
 
+        // win by killing president
+        var hasPresident = false;
```
Comment (pikulet) [inline Games/types/Mafia/roles/cards/WinWithMafia.js:55]: if there's 2 presidents does mafia win when killing one or both

**Comment by pikulet** [comment]
good work splitting it up, this should really be 3 PRs still but it's more manageable than your previous changes

don't have to change it, just fyi

File: Games/types/Mafia/roles/cards/Polariser.js (hunk 2 above)
Comment (Golbolco) [reply Games/types/Mafia/roles/cards/Polariser.js:11]: no, just multiactor

File: Games/types/Mafia/roles/cards/WinFourPolarisedDeaths.js (hunk 3 above)
Comment (Golbolco) [reply Games/types/Mafia/roles/cards/WinFourPolarisedDeaths.js:15]: what to use in place? this.player.role.name === "Polar Bear" would work?

File: Games/types/Mafia/roles/cards/WinWithMafia.js (hunk 4 above)
Comment (Golbolco) [reply Games/types/Mafia/roles/cards/WinWithMafia.js:55]: must kill only one to win

**Comment by Golbolco** [comment]
> good work splitting it up, this should really be 3 PRs still but it's more manageable than your previous changes
> 
> don't have to change it, just fyi

lol fair, each role = 1 pr from now on

File: Games/types/Mafia/roles/Village/President.js (hunk 1 above)
Comment (Golbolco) [reply Games/types/Mafia/roles/Village/President.js:9]: yes, let's use R9's instead then

**Review by pikulet — APPROVED** [review]

**Pull request merged**